src/peer_manager.rs

```diff
@@ -981,14 +982,33 @@ impl PeerManager {
             None
         };
 
+        let mut demote_peers = Vec::new();
         let expired_peers = self.peers
             .values()
-            .filter(|peer| peer.is_expired())
+            .filter(|peer| {
+                if peer.is_expired() {
```

Comment:
Will this even work ? If a peer is already in Peer-state of Routing, it will never be flagged as expired (that check should be bypassed else it would get removed below even in previous code) hence will never enter this, thus domotion logic will not work ?

Or does it work in some other way ?